cmd/skaffold/app/cmd/deploy.go

```diff
@@ -47,11 +49,35 @@ E.g. build.out created by running skaffold build --quiet {{json .}} > build.out`
 }
 
 func doDeploy(ctx context.Context, out io.Writer) error {
-	return withRunner(ctx, func(r runner.Runner, _ *latest.SkaffoldConfig) error {
-		// If the BuildArtifacts contains an image in the preBuilt list,
-		// use image from BuildArtifacts instead
-		deployArtifacts := build.MergeWithPreviousBuilds(buildOutputFile.BuildArtifacts(), preBuiltImages.Artifacts())
+	return withRunner(ctx, func(r runner.Runner, config *latest.SkaffoldConfig) error {
+		deployed, err := getDeployedArtifacts(out, buildOutputFile.BuildArtifacts(), preBuiltImages.Artifacts(), config.Build.Artifacts)
+		if err != nil {
+			return err
+		}
 
-		return r.DeployAndLog(ctx, out, deployArtifacts)
+		return r.DeployAndLog(ctx, out, deployed)
 	})
 }
+
+func getDeployedArtifacts(out io.Writer, fromFile, fromCLI []build.Artifact, artifacts []*latest.Artifact) ([]build.Artifact, error) {
```

Comment:
nit: I don't think these artifacts are already deployed: 
```suggestion
func getArtifactsToDeploy(out io.Writer, fromFile, fromCLI []build.Artifact, artifacts []*latest.Artifact) ([]build.Artifact, error) {
```